src/content/docs/dev/general-message-passing/soroban-gmp/soroban-contracts.mdx

```diff
@@ -0,0 +1,128 @@
+) -> Result<(), ContractError> {}
+```
+
+## Soroban Executable
+The [Executable](https://github.com/axelarnetwork/axelar-cgp-soroban/blob/main/contracts/axelar-gateway/src/executable.rs#L9) is a Rust Trait (not an individual contract) that will be used to execute a cross-chain call on a Soroban dapp when Stellar is the receiving chain of a cross-chain message.
+
+### Execute 
+The `execute()` function will be triggered by an Axelar relayer when the cross-chain message arrives. Your contract should implement the `execute()` function to handle the incoming cross-chain GMP data.
+
+The `execute()` function takes five parameters.
+1. `env`: Standard soroban [env](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/environment-concepts) that provides access to the environment the contract is executing within. 
+1. `source_chain`: The source chain where the gmp message is coming from.
+1. `message_id`: Identifier for incoming GMP message.
+1. `source_address`: The address on the source chain where the gmp message is coming from/
+1. `payload`: A `bytes` representation of the cross-chain message being sent.
+
+```rust
+fn execute(
+    env: Env,
+    source_chain: String,
+    message_id: String,
+    source_address: String,
+    payload: Bytes,
+);
+```
+
+### Validate & Validate Message
+The `validate()` function on the `Executable` will trigger the `validate_message()` that is [defined](https://github.com/axelarnetwork/axelar-cgp-soroban/blob/main/contracts/axelar-gateway/src/contract.rs#L128) on the `Gateway`.
+
+The `validate()` function takes five parameters.
+1. `env`: Standard soroban [env](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/environment-concepts) that provides access to the environment the contract is executing within. 
+1. `source_chain`: The source chain where the gmp message is coming from.
+1. `message_id`: Identifier for incoming GMP message.
+1. `source_address`: The address on the source chain where the gmp message is coming from/
+1. `payload`: A `bytes` representation of the cross-chain message being sent.
+
+```rust
+    fn validate(
+        env: Env,
+        source_chain: String,
+        message_id: String,
+        source_address: String,
+        payload: Bytes,
+    ) {}
+```
+
+The `validate()` function will trigger the `validate_message()`, this will confirm that the incoming GMP message is an authenticated message that has been verified by the Amplifier Verifiers.
+
+```rust
+     pub fn validate_message(
+        env: Env,
+        caller: Address,
+        source_chain: String,
+        message_id: String,
+        source_address: String,
+        payload_hash: BytesN<32>,
+    ) -> bool {}
+```
+
+1. `env`: Standard soroban [env](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/environment-concepts) that provides access to the environment the contract is executing within. 
+1. `caller`: (optional) The sender of the contract call. 
```

Comment:
In this case, `caller` must be the intended `destination_address` of the contract call